### PR TITLE
feat: improve Supervision CC usage, use automatically instead of hard-coded

### PR DESCRIPTION
**Pull request by AlCalzone**
The old CC specific hardcoded implementation was awkward and unflexible. The driver now automatically determines for all commands whether supervision can and should be used and does so automatically. As a result, `sendCommand` calls may now return a `SupervisionResult`.

fixes: #4716
fixes: https://github.com/zwave-js/node-zwave-js/issues/4497

### CODE REVIEW

File: packages/cc/src/cc/ConfigurationCC.ts [hunk 1]
```diff
@@ -416,22 +417,27 @@ export class ConfigurationCCAPI extends CCAPI {
...
 			parameter: property,
 			value: targetValue,
 			valueSize: valueSize as any,
 			valueFormat,
 		});
 
-		if ((this as ConfigurationCCAPI).isSinglecast()) {
-			// Verify the current value after a delay
+		if (
+			!isUnsupervisedOrSucceeded(result) &&
```
Comment (OXINARF) [inline packages/cc/src/cc/ConfigurationCC.ts]: Shouldn't this be `!supervisedCommandSucceeded(result)`? As it is, if it is unsupervised, it won't verify.

File: packages/cc/src/cc/BarrierOperatorCC.ts [hunk 2]
```diff
@@ -232,12 +236,14 @@ export class BarrierOperatorCCAPI extends CCAPI {
 				value === BarrierState.Closed
 					? BarrierState.Closed
 					: BarrierState.Open;
-			await this.set(targetValue);
+			const result = await this.set(targetValue);
 
-			// Verify the change after a delay
-			if (this.isSinglecast()) {
+			// Verify the change after a delay, unless the command was supervised and successful
+			if (!supervisedCommandSucceeded(result)) {
```
Comment (OXINARF) [inline packages/cc/src/cc/BarrierOperatorCC.ts]: Was it intended to remove the isSinglecast check?

File: packages/cc/src/cc/BasicCC.ts [hunk 3]
```diff
@@ -109,34 +112,39 @@ export class BasicCCAPI extends CCAPI {
 		if (typeof value !== "number") {
 			throwWrongValueType(this.ccId, property, "number", typeof value);
 		}
-		await this.set(value);
+
+		const result = await this.set(value);
+
+		// If the command did not fail, assume that it succeeded and update the currentValue accordingly
+		// so UIs have immediate feedback
```
Comment (OXINARF) [inline packages/cc/src/cc/BasicCC.ts]: Comment is duplicated below.

File: packages/cc/src/cc/MultilevelSwitchCC.ts [hunk 4]
```diff
@@ -393,81 +309,104 @@ export class MultilevelSwitchCCAPI extends CCAPI {
...
+									// eslint-disable-next-line @typescript-eslint/no-empty-function
+									void this.get().catch(() => {});
+								}
+							},
+					  }
+					: {},
+			).set(value, duration);
 
 			// If the command did not fail, assume that it succeeded and update the currentValue accordingly
 			// so UIs have immediate feedback
```
Comment (OXINARF) [inline packages/cc/src/cc/MultilevelSwitchCC.ts:347]: Comment is duplicated below.

**Review by OXINARF — COMMENTED** [review]
Just a couple of things I noticed while looking at the PR, sorry if it is just noise, I know this is still in draft.

**Comment by AlCalzone** [comment]
Thanks @OXINARF

**Pull request merged**